modules/apps/portal-security/portal-security-script-management-test/src/testIntegration/java/com/liferay/portal/security/script/management/web/internal/portlet/action/test/GetGroovyScriptUsesMVCResourceCommandTest.java

```diff
@@ -0,0 +1,347 @@
+			active, engine,
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			LocalizedMapUtil.getLocalizedMap(label),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION, script,
+			false, Collections.emptyList());
+	}
+
+	private ObjectDefinition _createObjectDefinition(long userId)
+		throws Exception {
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				userId, 0, false, false, false,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionTestUtil.getRandomName(), null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				false, ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
+				Arrays.asList(
+					new TextObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"textObjectField"
+					).objectFieldSettings(
+						Collections.emptyList()
+					).build()));
+
+		return _objectDefinitionLocalService.publishCustomObjectDefinition(
+			TestPropsValues.getUserId(),
+			objectDefinition.getObjectDefinitionId());
+	}
+
+	private void _createObjectDefinitionGroovyScriptUses(
+			String companyName, ObjectDefinition objectDefinition, long userId)
+		throws Exception {
+
+		_addObjectAction(
+			true, companyName + "ActiveGroovyObjectAction",
+			ObjectActionExecutorConstants.KEY_GROOVY, objectDefinition, userId);
+		_addObjectAction(
+			true, companyName + "ActiveWebhookObjectAction",
+			ObjectActionExecutorConstants.KEY_WEBHOOK, objectDefinition,
+			userId);
+		_addObjectAction(
+			false, companyName + "InactiveGroovyObjectAction",
+			ObjectActionExecutorConstants.KEY_GROOVY, objectDefinition, userId);
+
+		_addObjectValidationRule(
+			true, ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY,
+			companyName + "ActiveGroovyObjectValidation", objectDefinition,
+			userId);
+		_addObjectValidationRule(
+			true, ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
+			companyName + "ActiveDDMObjectValidation", objectDefinition,
+			userId);
+		_addObjectValidationRule(
+			false, ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY,
+			companyName + "InactiveGroovyObjectValidation", objectDefinition,
+			userId);
+	}
+
+	private String _getGroovyScriptUsesJSONArrayString() throws Exception {
+		MockLiferayResourceResponse mockLiferayResourceResponse =
+			new MockLiferayResourceResponse();
+
+		_mvcResourceCommand.serveResource(
+			_mockLiferayResourceRequest, mockLiferayResourceResponse);
+
+		ByteArrayOutputStream byteArrayOutputStream =
+			(ByteArrayOutputStream)
+				mockLiferayResourceResponse.getPortletOutputStream();
+
+		return byteArrayOutputStream.toString();
+	}
+
+	private String _getSourceURL(
+			Company company, long objectDefinitionId,
+			String screenNavigationCategoryKey)
+		throws Exception {
+
+		String url = StringBundler.concat(
```

Comment:
Can we reuse this logic from GroovyScriptUseSourceURLFactory class? This way we avoid code/logic duplication.

Reply:
Done [here](https://github.com/liferay-objects/liferay-portal/pull/3172/commits/605bb6d9ca507298cae2042ddf5e0334ea764b1d)